src/tests/test_hss_lms.cpp

```diff
@@ -0,0 +1,232 @@
+class HSS_LMS_Too_Short_Test final : public Test {
+      Test::Result test_too_short_signature() {
+         Test::Result result("LMS-HSS");
+
+         auto sk = Botan::create_private_key("HSS-LMS", Test::rng(), "Truncated(SHA-256,192),HW(5,8)");
+
+         Botan::PK_Signer signer(*sk, Test::rng(), "");
+         Botan::PK_Verifier verifier(*sk, "");
+
+         std::vector<uint8_t> mes = {0xde, 0xad, 0xbe, 0xef};
+
+         signer.update(mes);
+         auto valid_sig = signer.signature(Test::rng());
+         verifier.update(mes);
+         result.confirm("Entire signature is valid", verifier.check_signature(valid_sig.data(), valid_sig.size()));
+         for(size_t n = 0; n < valid_sig.size(); ++n) {
+            result.test_no_throw("Verification does not throw", [&]() {
+               verifier.update(mes);
+               bool valid = verifier.check_signature(valid_sig.data(), n);
+               result.confirm("Too short signature is invalid", !valid);
+            });
+         }
+
+         return result;
+      }
+
+      Test::Result test_too_short_private_key() {
+         Test::Result result("LMS-HSS");
+
+         // HSS_LMS_PublicKey::key_length()
+         auto sk = Botan::create_private_key("HSS-LMS", Test::rng(), "Truncated(SHA-256,192),HW(5,8)");
+
+         auto sk_bytes = sk->private_key_bits();
+         result.test_no_throw("Entire private key valid", [&]() {
+            Botan::HSS_LMS_PrivateKey key(sk_bytes);
+            BOTAN_UNUSED(key);
+         });
+         for(size_t n = 0; n < sk_bytes.size(); ++n) {
+            result.test_throws<Botan::Decoding_Error>("Partial private key invalid", [&]() {
+               std::span<const uint8_t> partial_key = {sk_bytes.data(), n};
+               Botan::HSS_LMS_PrivateKey key(partial_key);
+               BOTAN_UNUSED(key);
+            });
+         }
+         return result;
+      }
+
+      Test::Result test_too_short_public_key() {
+         Test::Result result("LMS-HSS");
+
+         // HSS_LMS_PublicKey::key_length()
+         auto sk = Botan::create_private_key("HSS-LMS", Test::rng(), "Truncated(SHA-256,192),HW(5,8)");
+
+         auto sk_bytes = sk->public_key_bits();
+         result.test_no_throw("Entire public key valid", [&]() {
+            Botan::HSS_LMS_PublicKey key(sk_bytes);
+            BOTAN_UNUSED(key);
+         });
+         for(size_t n = 0; n < sk_bytes.size(); ++n) {
+            result.test_throws<Botan::Decoding_Error>("Partial public key invalid", [&]() {
+               std::span<const uint8_t> partial_key = {sk_bytes.data(), n};
+               Botan::HSS_LMS_PublicKey key(partial_key);
+               BOTAN_UNUSED(key);
+            });
+         }
+         return result;
+      }
+
+      std::vector<Test::Result> run() final {
+         return {test_too_short_signature(), test_too_short_private_key(), test_too_short_public_key()};
+      }
+};
+
+// TODO: this currently only calls APIs not covered by other tests.
```

Comment:
Probably adapt this a little if https://github.com/randombit/botan/pull/3702 gets merged first.

Reply:
The PR is merged, so we should add LMS to it (after a rebase if that is not yet done)